mlflow/tracing/otel/translation/vercel_ai.py

```diff
@@ -0,0 +1,88 @@
+import json
+from typing import Any
+
+from mlflow.entities.span import SpanType
+from mlflow.tracing.otel.translation.base import OtelSchemaTranslator
+
+
+class VercelAITranslator(OtelSchemaTranslator):
+    """Translator for Vercel AI SDK spans."""
+
+    # https://ai-sdk.dev/docs/ai-sdk-core/telemetry#collected-data
+    INPUT_VALUE_KEYS = [
+        # generateText
+        "ai.prompt",
+        # tool call
+        "ai.toolCall.args",
+        # embed
+        "ai.value",
+        "ai.values",
+        # NB: generateText.doGenerate inputs/outputs are handled separately
+    ]
+    OUTPUT_VALUE_KEYS = [
+        # generateText
+        "ai.response.text",
+        # tool call
+        "ai.toolCall.result",
+        # generateObject
+        "ai.response.object",
+        # embed
+        "ai.embedding",
+        "ai.embeddings",
+    ]
+
+    SPAN_KIND_ATTRIBUTE_KEY = "ai.operationId"
+    SPAN_KIND_TO_MLFLOW_TYPE = {
+        "ai.generateText": SpanType.LLM,
+        "ai.generateText.doGenerate": SpanType.LLM,
+        "ai.toolCall": SpanType.TOOL,
+        "ai.streamText": SpanType.LLM,
+        "ai.streamText.doStream": SpanType.LLM,
+        "ai.generateObject": SpanType.LLM,
+        "ai.generateObject.doGenerate": SpanType.LLM,
+        "ai.streamObject": SpanType.LLM,
+        "ai.streamObject.doStream": SpanType.LLM,
+        "ai.embed": SpanType.EMBEDDING,
+        "ai.embed.doEmbed": SpanType.EMBEDDING,
+        "ai.embedMany": SpanType.EMBEDDING,
+    }
+
+    def get_input_value(self, attributes: dict[str, Any]) -> Any:
+        if self._is_chat_span(attributes):
+            inputs = self._unpack_attributes_with_prefix(attributes, "ai.prompt.")
+            if "tools" in inputs or {}:
```

Comment:
```suggestion
            if "tools" in inputs:
```
inputs should always be a dictionary?